### DISCUSSION TITLE
Create PnPContext behind company proxy

### DISCUSSION THREAD

**Discussion by anisiaalexandra** [pnp/pnpcore#1453] [Q&A]
Hi,

I have been experiencing some issues with a Function App that creates PnPContext while behind company proxy. I use my own network everything work correctly without any issues or timeouts. As soon as I test my Function on the network of the company (behind proxy) I receive the following error while trying to create the context:

Error occured while calling the Azure function: A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond. (<company>.sharepoint.com:443)

I have already configured in the local.settings.json the following:
"HTTP_PROXY": "http://proxy:80",
"HTTPS_PROXY": "http://proxy:80",
"ALL_PROXY": "http://proxy:80",
"NO_PROXY": "http://localhost:1234"

without any success.

Can or should I configure proxy somewhere else? And if yes, what and where?

Any idea is appreciated. Thank you.

**Comment by jansenbe** [comment]
@anisiaalexandra : we don't have explicit proxy support built in. What you can try is to configure the proxy via event variables support by the  .NET HttpClient class. See https://learn.microsoft.com/en-us/dotnet/api/system.net.http.httpclient.defaultproxy?view=net-8.0#remarks. Let me know if you get it to work, if so I'll include this in the docs
